sdks/python/apache_beam/metrics/cells.py

```diff
@@ -238,6 +277,74 @@ def mean(self):
     return float(self.data.sum)/self.data.count
 
 
+class GaugeResult(object):
+  def __init__(self, data):
+    self.data = data
+
+  def __eq__(self, other):
+    return self.data == other.data
```

Comment:
<!--new_thread; commit:3b2d3da5e97094e4521c80e07118d80343634698; resolved:0-->
Please add `isinstance()` guard to `__eq__` (https://stackoverflow.com/questions/390250/elegant-ways-to-support-equivalence-equality-in-python-classes).